mozilla/blurts-server
# Opt-out mechanism for emails

# References: 
**Jira**
- [MNTOR-1239](https://mozilla-hub.atlassian.net/browse/MNTOR-1039): Endpoint to get unsubscribe link
- [MNTOR-1290](https://mozilla-hub.atlassian.net/browse/MNTOR-1290): Migrate Opt-out mechanism for monthly summary emails to V1



# Description

Migrates the existing flow for users to unsubscribe from emails. Currently, we only support unsubscribing from the monthly unresolved breach reports.

# Screenshot

<img width="756" alt="pr-2889-screenshot" src="https://user-images.githubusercontent.com/13835474/224504329-df502ef5-ec5c-4bc9-b3c1-7bd5da2d9d94.png">

# How to test

- Visit `/preview/emails` and select the “Monthly unresolved breaches” template
- Click on “unsubscribe” in the email footer
- You should be directed to `/user/unsubscribe-monthly` and see a message that you successfully unsubscribe
- If the query param `token` is missing the user should be redirected to `/`

# Checklist (Definition of Done)
